Demisto-SDK docker (#7071)
demisto/dockerfiles: docker/demisto-sdk/README.md

# Demisto-SDK docker

> Run demisto-sdk commands from within a docker.

Reviewer: ```suggestion
Allows running demisto-sdk commands from within a docker container.
```

> You can use this image to run validations locally on your work environment or inside your ci.

Reviewer: ```suggestion
You can use this image to run demisto-sdk commands locally, or as part of a CI/CD process.
```

> You can use this image to run validations locally on your work environment or inside your ci.

Reviewer: ```suggestion
You can use this image to run demisto-sdk commands locally, or as part of a CI/CD process.
```

## The Content Repository

> To use the Demisto-SDK you should have a content-like repository (or use the official [content repository](https://github.com/demisto/content)).

Reviewer: ```suggestion
To use the Demisto-SDK, make sure you have a content-like repository with Cortex XSOAR content, in a structure that matches the official [XSOAR Content repo](https://github.com/demisto/content).
 
 ```

> You can generate one from this [template](https://github.com/demisto/content-external-template)

Reviewer: ```suggestion
Such a repository may be generated using the following [template](https://github.com/demisto/content-external-template)
```

## Mounts

> We use volume mount in our environment to run on local content repository.

Reviewer: ```suggestion
Demisto-sdk uses volume mounts, to run on local content repository.
```

end of the sentence is unclear

Author: It's not the demisto-sdk that uses the volume mounts.

> Please note that there's a performance issue with mount on MacOS and Windows.
> You can solve the performance issue by running on Linux/Windows WSL2 or by cloning your repository to the `/content` directory inside the container.

Reviewer: ```suggestion
Please note that mounting on MacOS and Windows may cause slowness. 

To ensure the best performance, please either:
- Use a linux machine
- Use [Windows WSL2](https://docs.microsoft.com/en-us/windows/wsl/install) 
- Clone your content repository to the `/content` directory inside the container.
```

do we explain how to do the cloning?

Author: If someone is working with a docker, he should be familiar with vcs, container etc. I think we should draw a line between where we support and where we are not.

## Examples

> (All examples from the [content](https://github.com/demisto/content) repository).

Reviewer: ```suggestion
(All examples use Cortex XSOAR's official [content repository](https://github.com/demisto/content)).
```

### Validate command

> 

Reviewer: ```suggestion
for more information about the validate command, please refer to its [documentation](https://github.com/demisto/demisto-sdk/blob/master/demisto_sdk/commands/validate/README.md) on the [demisto-sdk repo](https://github.com/demisto/demisto-sdk).
```

```sh
docker run -it --rm --mount type=bind,source="$(pwd)",target=/content demisto/demisto-sdk:<tag> demisto-sdk validate -i Packs/ipinfo/Integrations/ipinfo_v2
```

> #### Break down the command

Reviewer: ```suggestion
#### Breaking down command arguments
```

- docker run  

>     Creates a container and run a command

Reviewer: ```suggestion
    Creates a container (if one does not exist) and runs the following command inside it
```

- -it  

>     Keep the stdin open and tty

Reviewer: ```suggestion
    Keeps the `stdin` open and tty
```

`and tty`?

Author: https://en.wikipedia.org/wiki/Tty_(Unix)

- --rm  

>     Removes the docker container when done (Remove if you want to reuse the container)

Reviewer: ```suggestion
    Removes the docker container when done (ommit this part to re-use the container in the future)
```

- --mount type=bind,source="$(pwd)",target=/content  

>     Connect the pwd (assuming you're in content) to the content directory in the container

Reviewer: ```suggestion
    Connects the pwd (assuming you're in content) to the container's content directory
```

> - demisto/demisto-sdk:\<tag> (Replace the tag with locked version)  

Reviewer: where does one get the tag?

Author: docker hub

>     The docker image name  

Reviewer: huh?

Author: what is the line above

- demisto-sdk validate -i Packs/ipinfo/Integrations/ipinfo_v2

>     The command to run

Reviewer: ```suggestion
    The demisto-sdk command to be run inside the container
```

### Lint command

> If you want to run the `lint` command, you must connect the docker daemon (docker in docker)

Reviewer: ```suggestion
To run the `lint` command, connect the docker daemon (docker inside a docker), as lint itself is run inside a docker container. 
```

```sh

> docker run -it --rm --mount type=bind,source="$(pwd)",target=/content --mount source=/var/run/docker.sock,target=/var/run/docker.sock,type=bind jochman/demisto-sdk <demisto-sdk-command>

Reviewer: is it still `jochman/demisto-sdk`?

Author: nope, should update it as well.

```

> #### The mount park

Reviewer: park?

Author: part? :)

- --mount source=/var/run/docker.sock,target=/var/run/docker.sock,type=bind  

>     Mount the docker daemon in container to the outside (docker in docker)

Reviewer: ```suggestion
    Mounts the docker deamon container, to allow using docker commands from within a docker container.
```

> # Demisto-SDK docker

Reviewer 2: we should document the env vars that need to be modified on `docker run` for some of the commands

You can use this image to run demisto-sdk commands locally, or as part of a CI/CD process.

## The Content Repository

To use the Demisto-SDK, make sure you have a content-like repository with Cortex XSOAR content, in a structure that matches the official [XSOAR Content repo](https://github.com/demisto/content).
 
Such a repository may be generated using the following [template](https://github.com/demisto/content-external-template)

## Mounts

Demisto-sdk uses volume mounts, to run on local content repository.
Please note that mounting on MacOS and Windows may cause slowness. 

To ensure the best performance, please either:
- Use a linux machine
- Use [Windows WSL2](https://docs.microsoft.com/en-us/windows/wsl/install) 
- Clone your content repository to the `/content` directory inside the container.

## Examples

(All examples use Cortex XSOAR's official [content repository](https://github.com/demisto/content)).

### Validate command
for more information about the validate command, please refer to its [documentation](https://github.com/demisto/demisto-sdk/blob/master/demisto_sdk/commands/validate/README.md) on the [demisto-sdk repo](https://github.com/demisto/demisto-sdk).
```sh

> docker run -it --rm --mount type=bind,source="$(pwd)",target=/content demisto/demisto-sdk:<tag> demisto-sdk validate -i Packs/ipinfo/Integrations/ipinfo_v2

Author: I think that it should be up to the customer. :)

> docker run -it --rm --mount type=bind,source="$(pwd)",target=/content demisto/demisto-sdk:<tag> demisto-sdk validate -i Packs/ipinfo/Integrations/ipinfo_v2

Reviewer: will a read-only bind allow `format` calls?
